// @sparkdesignsystem/spark-angular/esm2015/lib/components/sprk-masthead/sprk-masthead-accordion-item/sprk-masthead-accordion-item.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
import * as _ from 'lodash';
import { toggleAnimations } from '../../sprk-toggle/sprk-toggle-animations';
let SprkMastheadAccordionItemComponent = class SprkMastheadAccordionItemComponent {
    constructor() {
        /**
         * If `true`, the Masthead accordion item
         * will be open when it is rendered.
         */
        this.isOpen = false;
        /**
         * The name of the icon to use
         * for the clickable close icon.
         */
        this.iconTypeClosed = 'chevron-up-circle-two-color';
        /**
         * The name of the icon to use
         * for the clickable open icon.
         */
        this.iconTypeOpen = 'chevron-up-circle-two-color';
        /**
         * @ignore
         */
        this.componentID = _.uniqueId();
        /**
         * @ignore
         */
        this.accordion_controls_id = `accordionHeading__${this.componentID}`;
        /**
         * @ignore
         */
        this.currentIconType = this.iconTypeClosed;
        /**
         * @ignore
         */
        this.iconStateClass = '';
        /**
         * @ignore
         */
        this.animState = 'closed';
    }
    /**
     * @ignore
     */
    accordionState() {
        this.isOpen === false
            ? (this.animState = 'closed')
            : (this.animState = 'open');
        this.isOpen === false
            ? (this.currentIconType = this.iconTypeClosed)
            : (this.currentIconType = this.iconTypeOpen);
        this.isOpen === false
            ? (this.iconStateClass = '')
            : (this.iconStateClass = 'sprk-c-Icon--open');
    }
    /**
     * @ignore
     */
    toggleAccordion(event) {
        event.preventDefault();
        this.isOpen = !this.isOpen;
        this.accordionState();
    }
    /**
     * @ignore
     */
    getClasses() {
        const classArray = [
            'sprk-c-MastheadAccordion__item',
            'sprk-u-Overflow--hidden'
        ];
        if (this.isOpen) {
            classArray.push('sprk-c-MastheadAccordion__item--open');
        }
        if (this.isActive) {
            classArray.push('sprk-c-MastheadAccordion__item--active');
        }
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(className => {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    }
    /**
     * @ignore
     */
    getHeadingClasses() {
        const classArray = ['sprk-c-MastheadAccordion__heading'];
        if (this.additionalHeadingClasses) {
            this.additionalHeadingClasses.split(' ').forEach(className => {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    }
    ngOnInit() {
        this.accordionState();
    }
};
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkMastheadAccordionItemComponent.prototype, "title", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkMastheadAccordionItemComponent.prototype, "analyticsString", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkMastheadAccordionItemComponent.prototype, "idString", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkMastheadAccordionItemComponent.prototype, "additionalClasses", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkMastheadAccordionItemComponent.prototype, "additionalHeadingClasses", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkMastheadAccordionItemComponent.prototype, "isOpen", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], SprkMastheadAccordionItemComponent.prototype, "isActive", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkMastheadAccordionItemComponent.prototype, "iconTypeClosed", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkMastheadAccordionItemComponent.prototype, "iconTypeOpen", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkMastheadAccordionItemComponent.prototype, "leadingIcon", void 0);
SprkMastheadAccordionItemComponent = __decorate([
    Component({
        selector: 'sprk-masthead-accordion-item',
        template: `
    <li [ngClass]="getClasses()">
      <div
        class="sprk-u-Position--relative sprk-o-Stack__item sprk-u-Width-100"
      >
        <button
          [attr.aria-controls]="accordion_controls_id"
          class="sprk-c-MastheadAccordion__summary"
          [attr.data-analytics]="analyticsString"
          [attr.data-id]="idString"
          (click)="toggleAccordion($event)"
          [attr.aria-expanded]="isOpen"
        >
          <span [ngClass]="getHeadingClasses()">
            <sprk-icon
              [iconType]="leadingIcon"
              additionalClasses="sprk-c-Icon--stroke-current-color sprk-c-Icon--l sprk-c-Icon--toggle sprk-u-mrs"
              *ngIf="leadingIcon"
            ></sprk-icon>
            {{ title }}
          </span>

          <sprk-icon
            additionalClasses="sprk-c-MastheadAccordion__icon sprk-c-Icon--stroke-current-color sprk-c-Icon--toggle {{
              iconStateClass
            }}"
            [iconType]="currentIconType"
          ></sprk-icon>
        </button>
      </div>

      <div [@toggleContent]="animState">
        <div [id]="accordion_controls_id"><ng-content></ng-content></div>
      </div>
    </li>
  `,
        animations: [toggleAnimations.toggleContent]
    })
], SprkMastheadAccordionItemComponent);
export { SprkMastheadAccordionItemComponent };
