// @sparkdesignsystem/spark-angular/esm2015/lib/directives/inputs/sprk-button/sprk-button.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, OnInit, Input, HostBinding, OnChanges, Renderer2, AfterViewInit, SimpleChanges, } from '@angular/core';
let SprkButtonDirective = class SprkButtonDirective {
    /**
     * @ignore
     */
    constructor(ref, renderer) {
        this.ref = ref;
        this.renderer = renderer;
        /**
         * Will show a spinner inside the
         * button if set to `true`.
         */
        this.isSpinning = false;
        /**
         * Optional string value that is
         * set for the `aria-label`
         * when `isSpinning` is `true`.
         */
        this.spinningAriaLabel = 'Loading';
        /**
         *  Determines the coresponding button style.
         */
        this.variant = 'primary';
        /**
         * @ignore
         */
        this.setSpinning = (element) => {
            const el = element;
            const width = element.offsetWidth;
            const ariaLabel = this.spinningAriaLabel;
            let spinnerClass = 'sprk-c-Spinner sprk-c-Spinner--circle';
            if (el.classList.contains('sprk-c-Button--secondary') ||
                this.variant === 'secondary') {
                spinnerClass += ' sprk-c-Spinner--dark';
            }
            this.renderer.setAttribute(el, 'data-sprk-spinner-text', el.textContent);
            this.renderer.setAttribute(el, 'aria-label', ariaLabel);
            el.innerHTML = `<div class="${spinnerClass}"></div>`;
            el.setAttribute('data-sprk-has-spinner', 'true');
            this.renderer.setAttribute(el, 'data-sprk-has-spinner', 'true');
            this.renderer.setAttribute(el, 'style', `width: ${width}px`);
        };
    }
    ngOnInit() {
        if (this.variant === 'secondary') {
            this.renderer.addClass(this.ref.nativeElement, 'sprk-c-Button--secondary');
        }
        if (this.variant === 'tertiary') {
            this.renderer.addClass(this.ref.nativeElement, 'sprk-c-Button--tertiary');
        }
    }
    /**
     * Add spinner only after view has loaded.
     * This is to allow time for the text
     * content of the button to load so that
     * the width value accounts for that text.
     */
    ngAfterViewInit() {
        if (this.isSpinning) {
            this.setSpinning(this.ref.nativeElement);
        }
    }
    ngOnChanges(changes) {
        if (this.isSpinning && !changes['isSpinning'].isFirstChange()) {
            this.setSpinning(this.ref.nativeElement);
        }
    }
};
SprkButtonDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkButtonDirective.prototype, "isSpinning", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkButtonDirective.prototype, "spinningAriaLabel", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkButtonDirective.prototype, "variant", void 0);
__decorate([
    HostBinding('class.sprk-c-Button'),
    __metadata("design:type", Object)
], SprkButtonDirective.prototype, "true", void 0);
__decorate([
    HostBinding('attr.data-id'),
    Input(),
    __metadata("design:type", String)
], SprkButtonDirective.prototype, "idString", void 0);
__decorate([
    HostBinding('attr.data-analytics'),
    Input(),
    __metadata("design:type", String)
], SprkButtonDirective.prototype, "analyticsString", void 0);
SprkButtonDirective = __decorate([
    Directive({
        selector: '[sprkButton]'
    }),
    __metadata("design:paramtypes", [ElementRef, Renderer2])
], SprkButtonDirective);
export { SprkButtonDirective };
