// @sparkdesignsystem/spark-angular/esm5/lib/components/sprk-highlight-board/sprk-highlight-board.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
var SprkHighlightBoardComponent = /** @class */ (function () {
    function SprkHighlightBoardComponent() {
        /**
         * @ignore
         */
        this.centerBtns = '';
    }
    SprkHighlightBoardComponent.prototype.ngOnInit = function () {
        if (!this.ctaText && this.ctaText2) {
            console.warn("\n        Spark Design System Warning - <sprk-highlight-board> Component:\n        A secondary call-to-action (CTA)\n        should not exist without a primary CTA.\n        If there is only one CTA, it must be set on\n        ctaText and ctaHref.\n      ");
        }
    };
    /**
     * @ignore
     */
    SprkHighlightBoardComponent.prototype.getClasses = function () {
        var classArray = ['sprk-c-HighlightBoard'];
        switch (this.type) {
            case 'noImage':
                this.centerBtns = 'sprk-o-Stack--center-row';
                break;
            case 'stacked':
                this.centerBtns = 'sprk-o-Stack--center-row';
                classArray.push('sprk-c-HighlightBoard--has-image sprk-c-HighlightBoard--stacked');
                break;
            default:
                classArray.push('sprk-c-HighlightBoard--has-image');
                break;
        }
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "heading", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "imgSrc", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "imgAlt", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "ctaText", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "ctaHref", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "ctaText2", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "ctaHref2", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "analyticsStringCta", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "analyticsStringCta2", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "additionalClasses", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "type", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkHighlightBoardComponent.prototype, "idString", void 0);
    SprkHighlightBoardComponent = __decorate([
        Component({
            selector: 'sprk-highlight-board',
            template: "\n    <div [ngClass]=\"getClasses()\" [attr.data-id]=\"idString\">\n      <img\n        *ngIf=\"type !== 'noImage'\"\n        class=\"sprk-c-HighlightBoard__image\"\n        src=\"{{ imgSrc }}\"\n        alt=\"{{ imgAlt }}\"\n      />\n\n      <sprk-stack\n        sprkStackItem\n        itemSpacing=\"large\"\n        additionalClasses=\"sprk-c-HighlightBoard__content\"\n      >\n        <h1\n          sprkStackItem\n          [ngClass]=\"{\n            'sprk-b-TypeDisplayTwo': type === 'noImage',\n            'sprk-b-TypeDisplayOne': type !== 'noImage',\n            'sprk-c-HighlightBoard__heading': true\n          }\"\n        >\n          {{ heading }}\n        </h1>\n\n        <sprk-stack\n          sprkStackItem\n          *ngIf=\"ctaText\"\n          itemSpacing=\"medium\"\n          splitAt=\"tiny\"\n          additionalClasses=\"sprk-o-Stack--center-column {{ centerBtns }}\"\n        >\n          <div sprkStackItem class=\"sprk-c-HighlightBoard__cta\">\n            <a\n              sprkLink\n              variant=\"unstyled\"\n              [attr.href]=\"ctaHref\"\n              class=\"sprk-c-Button sprk-c-Button--primary sprk-c-Button--full@sm\"\n              [analyticsString]=\"analyticsStringCta\"\n            >\n              {{ ctaText }}\n            </a>\n          </div>\n\n          <div\n            sprkStackItem\n            *ngIf=\"ctaText2\"\n            class=\"sprk-c-HighlightBoard__cta\"\n          >\n            <a\n              sprkLink\n              variant=\"unstyled\"\n              [attr.href]=\"ctaHref2\"\n              class=\"sprk-c-Button sprk-c-Button--secondary sprk-c-Button--full@sm\"\n              [analyticsString]=\"analyticsStringCta2\"\n            >\n              {{ ctaText2 }}\n            </a>\n          </div>\n        </sprk-stack>\n      </sprk-stack>\n    </div>\n  "
        })
    ], SprkHighlightBoardComponent);
    return SprkHighlightBoardComponent;
}());
export { SprkHighlightBoardComponent };
