// @sparkdesignsystem/spark-angular/esm5/lib/components/sprk-award/sprk-award.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
var SprkAwardComponent = /** @class */ (function () {
    function SprkAwardComponent() {
    }
    /**
     * @ignore
     */
    SprkAwardComponent.prototype.getClasses = function () {
        var classArray = [
            'sprk-o-Stack sprk-o-Stack--medium sprk-o-Stack__item sprk-o-Stack__item--center-column'
        ];
        // Handle the choice of item split
        // breakpoint by adding CSS class
        switch (this.splitAt) {
            case 'tiny':
                classArray.push('sprk-o-Stack--split@xs');
                break;
            case 'small':
                classArray.push('sprk-o-Stack--split@s');
                break;
            case 'medium':
                classArray.push('sprk-o-Stack--split@m');
                break;
            case 'large':
                classArray.push('sprk-o-Stack--split@l');
                break;
            case 'huge':
                classArray.push('sprk-o-Stack--split@xl');
                break;
            default:
                break;
        }
        return classArray.join(' ');
    };
    /**
     * @ignore
     */
    SprkAwardComponent.prototype.getImgContainerClasses = function () {
        var classArray = [
            'sprk-o-Stack__item'
        ];
        // Handle the choice of item split
        // breakpoint by adding CSS class
        switch (this.splitAt) {
            case 'tiny':
                classArray.push('sprk-o-Stack__item--flex@xs');
                break;
            case 'small':
                classArray.push('sprk-o-Stack__item--flex@s');
                break;
            case 'medium':
                classArray.push('sprk-o-Stack__item--flex@m');
                break;
            case 'large':
                classArray.push('sprk-o-Stack__item--flex@l');
                break;
            case 'huge':
                classArray.push('sprk-o-Stack__item--flex@xl');
                break;
            default:
                break;
        }
        return classArray.join(' ');
    };
    /**
     * @ignore
     */
    SprkAwardComponent.prototype.getClassesImgOne = function () {
        var classArray = [
            'sprk-o-Stack__item sprk-o-Stack__item--center-column'
        ];
        if (this.additionalClassesImgOne) {
            this.additionalClassesImgOne.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    /**
     * @ignore
     */
    SprkAwardComponent.prototype.getClassesImgTwo = function () {
        var classArray = [
            'sprk-o-Stack__item sprk-o-Stack__item--center-column'
        ];
        if (this.additionalClassesImgTwo) {
            this.additionalClassesImgTwo.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "splitAt", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "imgOneAlt", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "imgOneHref", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "imgTwoHref", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "imgTwoAlt", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "imgOneSrc", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "imgTwoSrc", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "analyticsStringImgOne", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "analyticsStringImgTwo", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "analyticsStringDisclaimer", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "additionalClasses", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "additionalClassesImgOne", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "additionalClassesImgTwo", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "title", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "disclaimerTitle", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "disclaimerCopy", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "disclaimer", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAwardComponent.prototype, "idString", void 0);
    SprkAwardComponent = __decorate([
        Component({
            selector: 'sprk-award',
            template: "\n    <sprk-stack\n      itemSpacing=\"medium\"\n      additionalClasses=\"{{ additionalClasses }}\"\n    >\n      <h2\n        sprkStackItem\n        [attr.data-id]=\"idString\"\n        class=\"\n          sprk-o-Stack__item\n          sprk-b-TypeDisplayFive\n          sprk-b-Measure sprk-b-Measure--narrow\n          sprk-u-TextAlign--center\n          sprk-o-Stack__item--center-column\n        \"\n      >\n        {{ title }}\n      </h2>\n\n      <div sprkStackItem [ngClass]=\"getClasses()\">\n        <div\n          [ngClass]=\"getImgContainerClasses()\"\n        >\n          <a\n            sprkLink\n            variant=\"unstyled\"\n            class=\"sprk-o-Stack\"\n            [attr.href]=\"imgOneHref\"\n            [analyticsString]=\"analyticsStringImgOne\"\n          >\n            <img\n              [ngClass]=\"getClassesImgOne()\"\n              alt=\"{{ imgOneAlt }}\"\n              src=\"{{ imgOneSrc }}\"\n            />\n          </a>\n        </div>\n        <div\n          [ngClass]=\"getImgContainerClasses()\"\n        >\n          <a\n            sprkLink\n            variant=\"unstyled\"\n            class=\"sprk-o-Stack\"\n            [attr.href]=\"imgTwoHref\"\n            [analyticsString]=\"analyticsStringImgTwo\"\n          >\n            <img\n              [ngClass]=\"getClassesImgTwo()\"\n              alt=\"{{ imgTwoAlt }}\"\n              src=\"{{ imgTwoSrc }}\"\n            />\n          </a>\n        </div>\n      </div>\n\n      <sprk-toggle\n        *ngIf=\"disclaimer !== 'false'\"\n        sprkStackItem\n        additionalClasses=\"sprk-o-Stack__item--start-column\"\n        toggleType=\"base\"\n        title=\"{{ disclaimerTitle }}\"\n        analyticsString=\"{{ analyticsStringDisclaimer }}\"\n      >\n        <p class=\"sprk-b-TypeBodyFour\">{{ disclaimerCopy }}</p>\n      </sprk-toggle>\n    </sprk-stack>\n  "
        })
    ], SprkAwardComponent);
    return SprkAwardComponent;
}());
export { SprkAwardComponent };
