// @sparkdesignsystem/spark-angular/esm5/lib/components/sprk-alert/sprk-alert.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
var SprkAlertComponent = /** @class */ (function () {
    function SprkAlertComponent() {
        /**
         * @ignore
         */
        this.visible = true;
    }
    /**
     * @ignore
     */
    SprkAlertComponent.prototype.getClassesAlertContainer = function () {
        var alertClassArray = ['sprk-c-Alert'];
        switch (this.alertType) {
            case 'success':
                alertClassArray.push('sprk-c-Alert--success');
                this.icon = 'check-mark';
                break;
            case 'info':
                alertClassArray.push('sprk-c-Alert--info');
                this.icon = 'bell';
                break;
            case 'fail':
                alertClassArray.push('sprk-c-Alert--fail');
                this.icon = 'exclamation';
                break;
            default:
                break;
        }
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(function (className) {
                alertClassArray.push(className);
            });
        }
        return alertClassArray.join(' ');
    };
    /**
     * When the dismiss button is clicked
     * this method sets the visable state to `false`
     * and hides the Alert component.
     */
    SprkAlertComponent.prototype.alertDismiss = function (event) {
        this.visible = false;
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAlertComponent.prototype, "alertType", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAlertComponent.prototype, "analyticsString", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAlertComponent.prototype, "idString", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkAlertComponent.prototype, "additionalClasses", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SprkAlertComponent.prototype, "dismissible", void 0);
    SprkAlertComponent = __decorate([
        Component({
            selector: 'sprk-alert',
            template: "\n    <div\n      *ngIf=\"visible\"\n      [ngClass]=\"getClassesAlertContainer()\"\n      role=\"alert\"\n      [attr.data-analytics]=\"analyticsString\"\n      [attr.data-id]=\"idString\"\n    >\n      <div class=\"sprk-c-Alert__content\">\n        <sprk-icon\n          iconType=\"{{ icon }}\"\n          additionalClasses=\"sprk-c-Alert__icon sprk-c-Icon--l sprk-c-Icon--stroke-current-color\"\n          aria-hidden=\"true\"\n        ></sprk-icon>\n\n        <p class=\"sprk-c-Alert__text\"><ng-content></ng-content></p>\n      </div>\n\n      <button\n        *ngIf=\"dismissible !== false\"\n        class=\"sprk-c-Alert__icon sprk-c-Alert__icon--dismiss\"\n        type=\"button\"\n        title=\"Dismiss\"\n        (click)=\"alertDismiss($event)\"\n      >\n        <sprk-icon\n          iconType=\"close\"\n          additionalClasses=\"sprk-c-Icon--stroke-current-color\"\n          aria-hidden=\"true\"\n        ></sprk-icon>\n      </button>\n    </div>\n  "
        })
    ], SprkAlertComponent);
    return SprkAlertComponent;
}());
export { SprkAlertComponent };
