// @sparkdesignsystem/spark-angular/esm2015/lib/directives/inputs/sprk-datepicker/sprk-datepicker.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnInit } from '@angular/core';
import TinyDatePicker from 'tiny-date-picker';
let SprkDatepickerDirective = class SprkDatepickerDirective {
    /**
     * @ignore
     */
    constructor(ref) {
        this.ref = ref;
    }
    ngOnInit() {
        const input = this.ref.nativeElement;
        const tdpConfig = {
            mode: 'dp-below',
            lang: {
                days: ['S', 'M', 'T', 'W', 'T', 'F', 'S']
            },
            min: '01/1/2008',
            max: '01/1/2068',
            format: date => date
                .toLocaleDateString('en-US', {
                month: '2-digit',
                day: '2-digit',
                year: 'numeric'
            })
                .replace(/[^ -~]/g, '')
        };
        if (TinyDatePicker) {
            TinyDatePicker(input, Object.assign(Object.assign({}, tdpConfig), this.sprkDatePickerConfig)).on('select', () => {
                let event;
                if (typeof (Event) === 'function') {
                    event = new Event('input');
                }
                else {
                    event = document.createEvent('Event');
                    event.initEvent('input', true, true);
                }
                this.ref.nativeElement.dispatchEvent(event);
                this.ref.nativeElement.focus();
            });
        }
    }
};
SprkDatepickerDirective.ctorParameters = () => [
    { type: ElementRef }
];
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkDatepickerDirective.prototype, "sprkDatePickerConfig", void 0);
SprkDatepickerDirective = __decorate([
    Directive({
        selector: '[sprkDatepicker]'
    }),
    __metadata("design:paramtypes", [ElementRef])
], SprkDatepickerDirective);
export { SprkDatepickerDirective };
