// @sparkdesignsystem/spark-angular/esm2015/lib/components/sprk-dictionary/sprk-dictionary.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
let SprkDictionaryComponent = class SprkDictionaryComponent {
    constructor() {
        /**
         * The collection of key-value pairs to be rendered
         * into the component.
         */
        this.data = {};
        /**
         * Determines the variant of the dictionary component to render.
         * The only available option is `striped`.
         * Supplying no value will cause the base variant to be used.
         */
        this.dictionaryType = 'base';
        /**
         * @ignore
         * Used to grab all the keys from
         * objects.
         */
        this.objectKeys = Object.keys;
    }
    /**
     * @ignore
     */
    getClasses() {
        const classArray = ['sprk-c-Dictionary'];
        if (this.dictionaryType === 'striped') {
            classArray.push('sprk-c-Dictionary--striped');
        }
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(className => {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    }
};
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkDictionaryComponent.prototype, "data", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], SprkDictionaryComponent.prototype, "dictionaryType", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkDictionaryComponent.prototype, "additionalClasses", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkDictionaryComponent.prototype, "idString", void 0);
SprkDictionaryComponent = __decorate([
    Component({
        selector: 'sprk-dictionary',
        template: `
    <div [ngClass]="getClasses()" [attr.data-id]="idString">
      <dl class="sprk-c-Dictionary__keyvaluepairs">
        <div
          *ngFor="let key of objectKeys(data)"
          class="sprk-c-Dictionary__keyvaluepair"
        >
          <dt
            class="sprk-c-Dictionary__key sprk-b-Label sprk-b-Label--no-input"
          >
            {{ key }}
          </dt>
          <dd class="sprk-c-Dictionary__value">{{ data[key] }}</dd>
        </div>
      </dl>
    </div>
  `
    })
], SprkDictionaryComponent);
export { SprkDictionaryComponent };
