// @sparkdesignsystem/spark-angular/esm5/lib/components/inputs/sprk-selection-container/sprk-selection-container.component.js
import { __decorate, __metadata } from "tslib";
import { Component, ContentChild, ContentChildren, Input, QueryList } from '@angular/core';
import * as _ from 'lodash';
import { SprkFieldErrorDirective } from '../../../directives/inputs/sprk-field-error/sprk-field-error.directive';
import { SprkLabelDirective } from '../../../directives/inputs/sprk-label/sprk-label.directive';
import { SprkSelectionItemContainerComponent } from '../sprk-selection-item-container/sprk-selection-item-container.component';
var SprkSelectionContainerComponent = /** @class */ (function () {
    function SprkSelectionContainerComponent() {
        /**
         * @ignore
         */
        this.id = _.uniqueId();
        /**
         * @ignore
         */
        this.error_id = "error_" + this.id;
    }
    /**
     * @ignore
     */
    SprkSelectionContainerComponent.prototype.getClasses = function () {
        var classArray = ['sprk-b-InputContainer'];
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    /**
     * @ignore
     */
    SprkSelectionContainerComponent.prototype.ngAfterContentInit = function () {
        var _this = this;
        if (this.selectionItems && this.error) {
            this.selectionItems.forEach(function (item) {
                item.input.ref.nativeElement.setAttribute('aria-describedby', _this.error_id);
            });
            this.error.ref.nativeElement.id = this.error_id;
        }
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkSelectionContainerComponent.prototype, "additionalClasses", void 0);
    __decorate([
        ContentChild(SprkLabelDirective, { static: false }),
        __metadata("design:type", SprkLabelDirective)
    ], SprkSelectionContainerComponent.prototype, "label", void 0);
    __decorate([
        ContentChild(SprkFieldErrorDirective, { static: false }),
        __metadata("design:type", SprkFieldErrorDirective)
    ], SprkSelectionContainerComponent.prototype, "error", void 0);
    __decorate([
        ContentChildren(SprkSelectionItemContainerComponent),
        __metadata("design:type", QueryList)
    ], SprkSelectionContainerComponent.prototype, "selectionItems", void 0);
    SprkSelectionContainerComponent = __decorate([
        Component({
            selector: 'sprk-selection-container',
            template: "\n    <div [ngClass]=\"getClasses()\">\n      <fieldset class=\"sprk-b-Fieldset\">\n        <legend class=\"sprk-b-Legend\">\n          <ng-content select=\"[sprkLabel]\"></ng-content>\n        </legend>\n        <ng-content select=\"sprk-selection-item-container\"></ng-content>\n      </fieldset>\n      <ng-content select=\"[sprkHelperText]\"></ng-content>\n      <ng-content select=\"[sprkFieldError]\"></ng-content>\n      <ng-content></ng-content>\n    </div>\n  "
        })
    ], SprkSelectionContainerComponent);
    return SprkSelectionContainerComponent;
}());
export { SprkSelectionContainerComponent };
