// @sparkdesignsystem/spark-angular/esm5/lib/components/sprk-toggle/sprk-toggle.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
import { toggleAnimations } from './sprk-toggle-animations';
import { uniqueId } from 'lodash';
import 'focus-visible';
var SprkToggleComponent = /** @class */ (function () {
    function SprkToggleComponent() {
        /**
         * The value supplied will be assigned as a CSS class
         * on the clickable title text used in the Toggle.
         * This is intended for overrides.
         */
        this.titleFontClass = 'sprk-b-TypeBodyThree';
        /**
         * A string that is used to set the `id` on the content
         * and the `aria-controls` for the toggle trigger button.
         */
        this.contentId = uniqueId("sprk_toggle_content_");
        /**
         * @ignore
         */
        this.isOpen = false;
        /**
         * @ignore
         */
        this.iconStateClass = '';
        /**
         * @ignore
         */
        this.animState = 'closed';
    }
    /**
     * @ignore
     */
    SprkToggleComponent.prototype.toggleState = function () {
        this.isOpen === false
            ? (this.animState = 'closed')
            : (this.animState = 'open');
        this.isOpen === false
            ? (this.iconStateClass = '')
            : (this.iconStateClass = 'sprk-c-Icon--open');
    };
    /**
     * @ignore
     */
    SprkToggleComponent.prototype.toggle = function (event) {
        event.preventDefault();
        this.isOpen = !this.isOpen;
        this.toggleState();
    };
    /**
     * @ignore
     */
    SprkToggleComponent.prototype.getClasses = function () {
        var classArray = [
            'sprk-c-Toggle__trigger sprk-u-TextCrop--none',
            this.titleFontClass,
        ];
        return classArray.join(' ');
    };
    SprkToggleComponent.prototype.ngAfterViewInit = function () {
        this.toggleState();
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkToggleComponent.prototype, "analyticsString", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkToggleComponent.prototype, "additionalClasses", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkToggleComponent.prototype, "title", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkToggleComponent.prototype, "iconClass", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], SprkToggleComponent.prototype, "titleFontClass", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkToggleComponent.prototype, "idString", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], SprkToggleComponent.prototype, "contentId", void 0);
    SprkToggleComponent = __decorate([
        Component({
            selector: 'sprk-toggle',
            template: "\n    <div\n      class=\"sprk-c-Toggle {{ additionalClasses }}\"\n      [attr.data-id]=\"idString\"\n    >\n      <button\n        variant=\"icon\"\n        [ngClass]=\"getClasses()\"\n        (click)=\"toggle($event)\"\n        [attr.aria-expanded]=\"isOpen ? 'true' : 'false'\"\n        [attr.data-analytics]=\"analyticsString\"\n        [attr.aria-controls]=\"contentId\"\n      >\n        <sprk-icon\n          iconType=\"chevron-down-circle-two-color\"\n          additionalClasses=\"{{\n            iconClass\n          }} sprk-c-Icon--l sprk-u-mrs sprk-c-Icon--toggle {{ iconStateClass }}\"\n        ></sprk-icon>\n        {{ title }}\n      </button>\n\n      <div\n        [@toggleContent]=\"animState\"\n        [id]=\"contentId\"\n      >\n        <div class=\"sprk-u-pts sprk-u-pbs sprk-c-Toggle__content\">\n          <ng-content></ng-content>\n        </div>\n      </div>\n    </div>\n  ",
            animations: [toggleAnimations.toggleContent]
        })
    ], SprkToggleComponent);
    return SprkToggleComponent;
}());
export { SprkToggleComponent };
