// @sparkdesignsystem/spark-angular/esm2015/lib/components/sprk-alert/sprk-alert.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
let SprkAlertComponent = class SprkAlertComponent {
    constructor() {
        /**
         * @ignore
         */
        this.visible = true;
    }
    /**
     * @ignore
     */
    getClassesAlertContainer() {
        const alertClassArray = ['sprk-c-Alert'];
        switch (this.alertType) {
            case 'success':
                alertClassArray.push('sprk-c-Alert--success');
                this.icon = 'check-mark';
                break;
            case 'info':
                alertClassArray.push('sprk-c-Alert--info');
                this.icon = 'bell';
                break;
            case 'fail':
                alertClassArray.push('sprk-c-Alert--fail');
                this.icon = 'exclamation';
                break;
            default:
                break;
        }
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(className => {
                alertClassArray.push(className);
            });
        }
        return alertClassArray.join(' ');
    }
    /**
     * When the dismiss button is clicked
     * this method sets the visable state to `false`
     * and hides the Alert component.
     */
    alertDismiss(event) {
        this.visible = false;
    }
};
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkAlertComponent.prototype, "alertType", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkAlertComponent.prototype, "analyticsString", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkAlertComponent.prototype, "idString", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkAlertComponent.prototype, "additionalClasses", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], SprkAlertComponent.prototype, "dismissible", void 0);
SprkAlertComponent = __decorate([
    Component({
        selector: 'sprk-alert',
        template: `
    <div
      *ngIf="visible"
      [ngClass]="getClassesAlertContainer()"
      role="alert"
      [attr.data-analytics]="analyticsString"
      [attr.data-id]="idString"
    >
      <div class="sprk-c-Alert__content">
        <sprk-icon
          iconType="{{ icon }}"
          additionalClasses="sprk-c-Alert__icon sprk-c-Icon--l sprk-c-Icon--stroke-current-color"
          aria-hidden="true"
        ></sprk-icon>

        <p class="sprk-c-Alert__text"><ng-content></ng-content></p>
      </div>

      <button
        *ngIf="dismissible !== false"
        class="sprk-c-Alert__icon sprk-c-Alert__icon--dismiss"
        type="button"
        title="Dismiss"
        (click)="alertDismiss($event)"
      >
        <sprk-icon
          iconType="close"
          additionalClasses="sprk-c-Icon--stroke-current-color"
          aria-hidden="true"
        ></sprk-icon>
      </button>
    </div>
  `
    })
], SprkAlertComponent);
export { SprkAlertComponent };
