// @sparkdesignsystem/spark-angular/esm2015/lib/directives/sprk-heading/sprk-heading.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, Input, HostBinding } from '@angular/core';
let SprkHeadingDirective = class SprkHeadingDirective {
    /**
     * @ignore
     */
    constructor() {
        this.isPageTitle = false;
    }
    get headingOne() {
        return this.variant === 'displayOne';
    }
    get headingTwo() {
        return this.variant === 'displayTwo';
    }
    get headingThree() {
        return this.variant === 'displayThree';
    }
    get headingFour() {
        return this.variant === 'displayFour';
    }
    get headingFive() {
        return this.variant === 'displayFive';
    }
    get headingSix() {
        return this.variant === 'displaySix';
    }
    get headingSeven() {
        return this.variant === 'displaySeven';
    }
};
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkHeadingDirective.prototype, "variant", void 0);
__decorate([
    HostBinding('class.sprk-b-TypeDisplayOne'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective.prototype, "headingOne", null);
__decorate([
    HostBinding('class.sprk-b-TypeDisplayTwo'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective.prototype, "headingTwo", null);
__decorate([
    HostBinding('class.sprk-b-TypeDisplayThree'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective.prototype, "headingThree", null);
__decorate([
    HostBinding('class.sprk-b-TypeDisplayFour'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective.prototype, "headingFour", null);
__decorate([
    HostBinding('class.sprk-b-TypeDisplayFive'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective.prototype, "headingFive", null);
__decorate([
    HostBinding('class.sprk-b-TypeDisplaySix'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective.prototype, "headingSix", null);
__decorate([
    HostBinding('class.sprk-b-TypeDisplaySeven'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective.prototype, "headingSeven", null);
__decorate([
    HostBinding('class.sprk-b-PageTitle')
    /**
     * Value will be used to select
     * the heading style.
     */
    ,
    Input(),
    __metadata("design:type", Object)
], SprkHeadingDirective.prototype, "isPageTitle", void 0);
SprkHeadingDirective = __decorate([
    Directive({
        selector: '[sprkHeading]'
    }),
    __metadata("design:paramtypes", [])
], SprkHeadingDirective);
export { SprkHeadingDirective };
