// @sparkdesignsystem/spark-angular/esm2015/lib/directives/sprk-box/sprk-box.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, Input, HostBinding } from '@angular/core';
let SprkBoxDirective = class SprkBoxDirective {
    /**
     * @ignore
     */
    constructor() { }
    get flush() {
        return this.spacing === 'flush';
    }
    get tiny() {
        return this.spacing === 'tiny';
    }
    get small() {
        return this.spacing === 'small';
    }
    get medium() {
        return this.spacing === 'medium';
    }
    get large() {
        return this.spacing === 'large';
    }
    get huge() {
        return this.spacing === 'huge';
    }
    get insetShortTiny() {
        return this.spacing === 'insetShortTiny';
    }
    get insetShortSmall() {
        return this.spacing === 'insetShortSmall';
    }
    get insetShortMedium() {
        return this.spacing === 'insetShortMedium';
    }
    get insetShortLarge() {
        return this.spacing === 'insetShortLarge';
    }
    get insetShortHuge() {
        return this.spacing === 'insetShortHuge';
    }
    get insetTallTiny() {
        return this.spacing === 'insetTallTiny';
    }
    get insetTallSmall() {
        return this.spacing === 'insetTallSmall';
    }
    get insetTallMedium() {
        return this.spacing === 'insetTallMedium';
    }
    get insetTallLarge() {
        return this.spacing === 'insetTallLarge';
    }
    get insetTallHuge() {
        return this.spacing === 'insetTallHuge';
    }
    get miscA() {
        return this.spacing === 'miscA';
    }
    get miscB() {
        return this.spacing === 'miscB';
    }
    get miscC() {
        return this.spacing === 'miscC';
    }
    get miscD() {
        return this.spacing === 'miscD';
    }
};
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkBoxDirective.prototype, "spacing", void 0);
__decorate([
    HostBinding('class.sprk-o-Box'),
    __metadata("design:type", Object)
], SprkBoxDirective.prototype, "true", void 0);
__decorate([
    HostBinding('class.sprk-o-Box--flush'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "flush", null);
__decorate([
    HostBinding('class.sprk-o-Box--tiny'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "tiny", null);
__decorate([
    HostBinding('class.sprk-o-Box--small'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "small", null);
__decorate([
    HostBinding('class.sprk-o-Box--medium'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "medium", null);
__decorate([
    HostBinding('class.sprk-o-Box--large'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "large", null);
__decorate([
    HostBinding('class.sprk-o-Box--huge'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "huge", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-short-tiny'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetShortTiny", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-short-small'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetShortSmall", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-short-medium'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetShortMedium", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-short-large'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetShortLarge", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-short-huge'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetShortHuge", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-tall-tiny'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetTallTiny", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-tall-small'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetTallSmall", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-tall-medium'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetTallMedium", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-tall-large'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetTallLarge", null);
__decorate([
    HostBinding('class.sprk-o-Box--inset-tall-huge'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "insetTallHuge", null);
__decorate([
    HostBinding('class.sprk-o-Box--misc-a'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "miscA", null);
__decorate([
    HostBinding('class.sprk-o-Box--misc-b'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "miscB", null);
__decorate([
    HostBinding('class.sprk-o-Box--misc-c'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "miscC", null);
__decorate([
    HostBinding('class.sprk-o-Box--misc-d'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkBoxDirective.prototype, "miscD", null);
SprkBoxDirective = __decorate([
    Directive({
        selector: '[sprkBox]'
    }),
    __metadata("design:paramtypes", [])
], SprkBoxDirective);
export { SprkBoxDirective };
