// @sparkdesignsystem/spark-angular/esm2015/lib/components/sprk-promo/sprk-promo.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
let SprkPromoComponent = class SprkPromoComponent {
    /**
     * @ignore
     */
    getClasses() {
        const classArray = [
            'sprk-c-Promo',
            'sprk-o-Stack',
            'sprk-o-Stack--split@s'
        ];
        if (this.isFlag) {
            classArray.push('sprk-c-Promo--flag');
        }
        if (this.hasBorder) {
            classArray.push('sprk-c-Promo--bordered');
        }
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(className => {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    }
    /**
     * @ignore
     */
    getClassesFlag() {
        const classArray = [
            'sprk-o-Stack__item--fourth@s',
            'sprk-o-Stack__item'
        ];
        if (this.additionalClassesFlagLink) {
            this.additionalClassesFlagLink.split(' ').forEach(className => {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    }
    /**
     * @ignore
     */
    getClassesImg() {
        const classArray = [
            'sprk-o-Stack__item',
            'sprk-o-Stack__item--half@s'
        ];
        if (this.additionalClassesImgLink) {
            this.additionalClassesImgLink.split(' ').forEach(className => {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    }
    /**
     * @ignore
     */
    getClassesContent() {
        const classArray = [
            'sprk-c-Promo__content',
            'sprk-o-Stack__item',
            'sprk-o-Stack',
            'sprk-o-Stack--large'
        ];
        if (this.isFlag) {
            classArray.push('sprk-o-Stack__item--three-fourths@s');
        }
        else {
            if (this.imgSrc) {
                classArray.push('sprk-o-Stack__item--half@s');
            }
        }
        if (this.additionalClassesContent) {
            this.additionalClassesContent.split(' ').forEach(className => {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    }
};
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "title", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "subtitle", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "cta", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "ctaText", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "buttonHref", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "buttonLinkAnalytics", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "ctaLinkHref", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "ctaLinkAnalytics", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "additionalClasses", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "additionalClassesContent", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], SprkPromoComponent.prototype, "isFlag", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "additionalClassesFlagLink", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "flagLinkAnalytics", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "imgSrc", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "imgAlt", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "imgHref", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "imgLinkAnalytics", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "additionalClassesImgLink", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], SprkPromoComponent.prototype, "mediaRev", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], SprkPromoComponent.prototype, "hasBorder", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkPromoComponent.prototype, "idString", void 0);
SprkPromoComponent = __decorate([
    Component({
        selector: 'sprk-promo',
        template: `
    <div [ngClass]="getClasses()" [attr.data-id]="idString">
      <a
        sprkLink
        variant="unstyled"
        *ngIf="isFlag && !mediaRev"
        [ngClass]="getClassesFlag()"
        [attr.href]="imgHref"
        [analyticsString]="flagLinkAnalytics"
      >
        <img
          class="sprk-c-Promo__image--flag"
          alt="{{ imgAlt }}"
          src="{{ imgSrc }}"
        />
      </a>
      <a
        sprkLink
        variant="unstyled"
        *ngIf="imgSrc && !mediaRev && !isFlag"
        [attr.href]="imgHref"
        [analyticsString]="imgLinkAnalytics"
        [ngClass]="getClassesImg()"
      >
        <img
          class="sprk-c-Promo__image"
          alt="{{ imgAlt }}"
          src="{{ imgSrc }}"
        />
      </a>

      <div [ngClass]="getClassesContent()">
        <h3
          *ngIf="title"
          class="sprk-b-TypeDisplayFive sprk-o-Stack__item sprk-c-Promo__title"
        >
          {{ title }}
        </h3>

        <div *ngIf="subtitle" class="sprk-c-Promo__subtitle sprk-o-Stack__item">
          {{ subtitle }}
        </div>

        <div class="sprk-b-TypeBodyTwo sprk-o-Stack__item">
          <ng-content></ng-content>
        </div>

        <div *ngIf="cta === 'button'" class="sprk-o-Stack__item">
          <a
            sprkLink
            variant="unstyled"
            [attr.href]="buttonHref"
            class="sprk-c-Button"
            [analyticsString]="buttonLinkAnalytics"
          >
            {{ ctaText }}
          </a>
        </div>

        <a
          sprkLink
          variant="simple"
          *ngIf="cta === 'link'"
          [attr.href]="ctaLinkHref"
          class="sprk-o-Stack__item"
          [analyticsString]="ctaLinkAnalytics"
        >
          {{ ctaText }}
        </a>
      </div>

      <a
        sprkLink
        variant="unstyled"
        *ngIf="isFlag && mediaRev"
        [attr.href]="imgHref"
        [ngClass]="getClassesFlag()"
        [analyticsString]="flagLinkAnalytics"
      >
        <img
          class="sprk-c-Promo__image--flag"
          alt="{{ imgAlt }}"
          src="{{ imgSrc }}"
        />
      </a>

      <a
        sprkLink
        variant="unstyled"
        *ngIf="imgSrc && mediaRev && !isFlag"
        [attr.href]="imgHref"
        [analyticsString]="imgLinkAnalytics"
        [ngClass]="getClassesImg()"
      >
        <img
          class="sprk-c-Promo__image"
          alt="{{ imgAlt }}"
          src="{{ imgSrc }}"
        />
      </a>
    </div>
  `
    })
], SprkPromoComponent);
export { SprkPromoComponent };
