// @sparkdesignsystem/spark-angular/esm5/lib/components/sprk-promo/sprk-promo.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
var SprkPromoComponent = /** @class */ (function () {
    function SprkPromoComponent() {
    }
    /**
     * @ignore
     */
    SprkPromoComponent.prototype.getClasses = function () {
        var classArray = [
            'sprk-c-Promo',
            'sprk-o-Stack',
            'sprk-o-Stack--split@s'
        ];
        if (this.isFlag) {
            classArray.push('sprk-c-Promo--flag');
        }
        if (this.hasBorder) {
            classArray.push('sprk-c-Promo--bordered');
        }
        if (this.additionalClasses) {
            this.additionalClasses.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    /**
     * @ignore
     */
    SprkPromoComponent.prototype.getClassesFlag = function () {
        var classArray = [
            'sprk-o-Stack__item--fourth@s',
            'sprk-o-Stack__item'
        ];
        if (this.additionalClassesFlagLink) {
            this.additionalClassesFlagLink.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    /**
     * @ignore
     */
    SprkPromoComponent.prototype.getClassesImg = function () {
        var classArray = [
            'sprk-o-Stack__item',
            'sprk-o-Stack__item--half@s'
        ];
        if (this.additionalClassesImgLink) {
            this.additionalClassesImgLink.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    /**
     * @ignore
     */
    SprkPromoComponent.prototype.getClassesContent = function () {
        var classArray = [
            'sprk-c-Promo__content',
            'sprk-o-Stack__item',
            'sprk-o-Stack',
            'sprk-o-Stack--large'
        ];
        if (this.isFlag) {
            classArray.push('sprk-o-Stack__item--three-fourths@s');
        }
        else {
            if (this.imgSrc) {
                classArray.push('sprk-o-Stack__item--half@s');
            }
        }
        if (this.additionalClassesContent) {
            this.additionalClassesContent.split(' ').forEach(function (className) {
                classArray.push(className);
            });
        }
        return classArray.join(' ');
    };
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "title", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "subtitle", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "cta", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "ctaText", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "buttonHref", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "buttonLinkAnalytics", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "ctaLinkHref", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "ctaLinkAnalytics", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "additionalClasses", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "additionalClassesContent", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SprkPromoComponent.prototype, "isFlag", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "additionalClassesFlagLink", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "flagLinkAnalytics", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "imgSrc", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "imgAlt", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "imgHref", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "imgLinkAnalytics", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "additionalClassesImgLink", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SprkPromoComponent.prototype, "mediaRev", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], SprkPromoComponent.prototype, "hasBorder", void 0);
    __decorate([
        Input(),
        __metadata("design:type", String)
    ], SprkPromoComponent.prototype, "idString", void 0);
    SprkPromoComponent = __decorate([
        Component({
            selector: 'sprk-promo',
            template: "\n    <div [ngClass]=\"getClasses()\" [attr.data-id]=\"idString\">\n      <a\n        sprkLink\n        variant=\"unstyled\"\n        *ngIf=\"isFlag && !mediaRev\"\n        [ngClass]=\"getClassesFlag()\"\n        [attr.href]=\"imgHref\"\n        [analyticsString]=\"flagLinkAnalytics\"\n      >\n        <img\n          class=\"sprk-c-Promo__image--flag\"\n          alt=\"{{ imgAlt }}\"\n          src=\"{{ imgSrc }}\"\n        />\n      </a>\n      <a\n        sprkLink\n        variant=\"unstyled\"\n        *ngIf=\"imgSrc && !mediaRev && !isFlag\"\n        [attr.href]=\"imgHref\"\n        [analyticsString]=\"imgLinkAnalytics\"\n        [ngClass]=\"getClassesImg()\"\n      >\n        <img\n          class=\"sprk-c-Promo__image\"\n          alt=\"{{ imgAlt }}\"\n          src=\"{{ imgSrc }}\"\n        />\n      </a>\n\n      <div [ngClass]=\"getClassesContent()\">\n        <h3\n          *ngIf=\"title\"\n          class=\"sprk-b-TypeDisplayFive sprk-o-Stack__item sprk-c-Promo__title\"\n        >\n          {{ title }}\n        </h3>\n\n        <div *ngIf=\"subtitle\" class=\"sprk-c-Promo__subtitle sprk-o-Stack__item\">\n          {{ subtitle }}\n        </div>\n\n        <div class=\"sprk-b-TypeBodyTwo sprk-o-Stack__item\">\n          <ng-content></ng-content>\n        </div>\n\n        <div *ngIf=\"cta === 'button'\" class=\"sprk-o-Stack__item\">\n          <a\n            sprkLink\n            variant=\"unstyled\"\n            [attr.href]=\"buttonHref\"\n            class=\"sprk-c-Button\"\n            [analyticsString]=\"buttonLinkAnalytics\"\n          >\n            {{ ctaText }}\n          </a>\n        </div>\n\n        <a\n          sprkLink\n          variant=\"simple\"\n          *ngIf=\"cta === 'link'\"\n          [attr.href]=\"ctaLinkHref\"\n          class=\"sprk-o-Stack__item\"\n          [analyticsString]=\"ctaLinkAnalytics\"\n        >\n          {{ ctaText }}\n        </a>\n      </div>\n\n      <a\n        sprkLink\n        variant=\"unstyled\"\n        *ngIf=\"isFlag && mediaRev\"\n        [attr.href]=\"imgHref\"\n        [ngClass]=\"getClassesFlag()\"\n        [analyticsString]=\"flagLinkAnalytics\"\n      >\n        <img\n          class=\"sprk-c-Promo__image--flag\"\n          alt=\"{{ imgAlt }}\"\n          src=\"{{ imgSrc }}\"\n        />\n      </a>\n\n      <a\n        sprkLink\n        variant=\"unstyled\"\n        *ngIf=\"imgSrc && mediaRev && !isFlag\"\n        [attr.href]=\"imgHref\"\n        [analyticsString]=\"imgLinkAnalytics\"\n        [ngClass]=\"getClassesImg()\"\n      >\n        <img\n          class=\"sprk-c-Promo__image\"\n          alt=\"{{ imgAlt }}\"\n          src=\"{{ imgSrc }}\"\n        />\n      </a>\n    </div>\n  "
        })
    ], SprkPromoComponent);
    return SprkPromoComponent;
}());
export { SprkPromoComponent };
