// @sparkdesignsystem/spark-angular/esm5/lib/directives/inputs/sprk-datepicker/sprk-datepicker.directive.js
import { __assign, __decorate, __metadata } from "tslib";
import { Directive, ElementRef, Input, OnInit } from '@angular/core';
import TinyDatePicker from 'tiny-date-picker';
var SprkDatepickerDirective = /** @class */ (function () {
    /**
     * @ignore
     */
    function SprkDatepickerDirective(ref) {
        this.ref = ref;
    }
    SprkDatepickerDirective.prototype.ngOnInit = function () {
        var _this = this;
        var input = this.ref.nativeElement;
        var tdpConfig = {
            mode: 'dp-below',
            lang: {
                days: ['S', 'M', 'T', 'W', 'T', 'F', 'S']
            },
            min: '01/1/2008',
            max: '01/1/2068',
            format: function (date) {
                return date
                    .toLocaleDateString('en-US', {
                    month: '2-digit',
                    day: '2-digit',
                    year: 'numeric'
                })
                    .replace(/[^ -~]/g, '');
            }
        };
        if (TinyDatePicker) {
            TinyDatePicker(input, __assign(__assign({}, tdpConfig), this.sprkDatePickerConfig)).on('select', function () {
                var event;
                if (typeof (Event) === 'function') {
                    event = new Event('input');
                }
                else {
                    event = document.createEvent('Event');
                    event.initEvent('input', true, true);
                }
                _this.ref.nativeElement.dispatchEvent(event);
                _this.ref.nativeElement.focus();
            });
        }
    };
    SprkDatepickerDirective.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], SprkDatepickerDirective.prototype, "sprkDatePickerConfig", void 0);
    SprkDatepickerDirective = __decorate([
        Directive({
            selector: '[sprkDatepicker]'
        }),
        __metadata("design:paramtypes", [ElementRef])
    ], SprkDatepickerDirective);
    return SprkDatepickerDirective;
}());
export { SprkDatepickerDirective };
