// @sparkdesignsystem/spark-angular/esm2015/lib/directives/sprk-text/sprk-text.directive.js
import { __decorate, __metadata } from "tslib";
import { Directive, Input, HostBinding } from '@angular/core';
let SprkTextDirective = class SprkTextDirective {
    /**
     * @ignore
     */
    constructor() { }
    get textOne() {
        return this.variant === 'bodyOne';
    }
    get textTwo() {
        return this.variant === 'bodyTwo';
    }
    get textThree() {
        return this.variant === 'bodyThree';
    }
    get textFour() {
        return this.variant === 'bodyFour';
    }
};
__decorate([
    Input(),
    __metadata("design:type", String)
], SprkTextDirective.prototype, "variant", void 0);
__decorate([
    HostBinding('class.sprk-b-TypeBodyOne'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkTextDirective.prototype, "textOne", null);
__decorate([
    HostBinding('class.sprk-b-TypeBodyTwo'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkTextDirective.prototype, "textTwo", null);
__decorate([
    HostBinding('class.sprk-b-TypeBodyThree'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkTextDirective.prototype, "textThree", null);
__decorate([
    HostBinding('class.sprk-b-TypeBodyFour'),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [])
], SprkTextDirective.prototype, "textFour", null);
SprkTextDirective = __decorate([
    Directive({
        selector: '[sprkText]'
    }),
    __metadata("design:paramtypes", [])
], SprkTextDirective);
export { SprkTextDirective };
